process/resultsParser/resultsParser.go

```diff
@@ -0,0 +1,401 @@
+
+			if strings.HasPrefix(string(topic), TooMuchGas) {
+				return topic
+			}
+			return nil
+		})
+
+		if t != nil {
+			return e
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to find single or none event: %w", err)
+	}
+
+	if event == nil {
+		return nil, nil
+	}
+
+	returnCode, returnDataParts, err := sliceDataFieldInParts(string(event.Data))
+	if err != nil {
+		return nil, fmt.Errorf("failed to slice data field in parts: %w", err)
+	}
+
+	lastTopic := getLastTopic(event.Topics)
+
+	returnMessage := returnCode.String()
+	if lastTopic != nil {
+		returnMessage = string(lastTopic)
+	}
+
+	return &ResultOutcome{
+		ReturnCode:    *returnCode,
+		ReturnMessage: returnMessage,
+		Values:        returnDataParts,
+	}, nil
+}
+
+func parseOutcomeOnWriteLogWhereFirstTopicEqualsAddress(logs *transaction.ApiLogs, address string) (*ResultOutcome, error) {
+	base64Address := base64.StdEncoding.EncodeToString([]byte(address))
+
+	event, err := findSingleOrNoneEvent(logs, OnWriteLog, func(e *transaction.Events) *transaction.Events {
+		t := findFirstOrNoneTopic(e.Topics, func(topic []byte) []byte {
+			if string(topic) == base64Address {
+				return topic
+			}
+			return nil
+		})
+
+		if t != nil {
+			return e
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to find single or none event: %w", err)
+	}
+
+	if event == nil {
+		return nil, nil
+	}
+
+	returnCode, returnDataParts, err := sliceDataFieldInParts(string(event.Data))
+	if err != nil {
+		return nil, fmt.Errorf("failed to slice data field in parts: %w", err)
+	}
+
+	return &ResultOutcome{
+		ReturnCode:    *returnCode,
+		ReturnMessage: returnCode.String(),
+		Values:        returnDataParts,
+	}, nil
+}
+
+func parseOutcomeWithFallbackHeuristics(tx *transaction.ApiTransactionResult, receiver string) (*ResultOutcome, error) {
+	for _, resultItem := range tx.SmartContractResults {
+		event, findErr := findSingleOrNoneEvent(resultItem.Logs, OnWriteLog, func(e *transaction.Events) *transaction.Events {
+			addressIsSender := e.Address == tx.Sender
+			firstTopicIsContract := false
+			if e.Topics[0] != nil {
+				decodeString, topicDecode := hex.DecodeString(string(e.Topics[0]))
+				if topicDecode != nil {
+					return nil
+				}
+				firstTopicIsContract = string(decodeString) == receiver
+
+				if firstTopicIsContract && addressIsSender {
+					return e
+				}
+			}
+
+			return nil
+		})
+		if findErr != nil {
+			return nil, findErr
+		}
+
+		if event == nil {
+			return nil, nil
+		}
+
+		returnCode, returnDataParts, sliceErr := sliceDataFieldInParts(string(event.Data))
+		if sliceErr != nil {
+			return nil, sliceErr
+		}
+
+		return &ResultOutcome{
+			ReturnCode:    *returnCode,
+			ReturnMessage: returnCode.String(),
+			Values:        returnDataParts,
+		}, nil
+	}
+
+	return nil, nil
+}
+
+func sliceDataFieldInParts(data string) (*ReturnCode, []*bytes.Buffer, error) {
```

Comment:
your data field is already parsed, if you fetch the transaction with results, no need for this function

Reply:
you should remove this function

Reply:
check this code and where is used 
https://github.com/multiversx/mx-chain-vm-common-go/tree/master/parsers/dataField

Reply:
`NewOperationDataFieldParser` cannot slice `SCR.data` and `event.data` (and this is what we want in this context).

```

func TestForProcessOutcome(t *testing.T) {
	arguments := createMockArgumentsOperationParser()
	dataParser, _ := NewOperationDataFieldParser(arguments)
	argsParser := parsers.NewCallArgsParser()

	dataField := []byte("@6f6b@01")

	parsedData := dataParser.Parse(dataField, sender, sender, 3)
	// Nothing, since "ResponseParseData" cannot hold what we want.
	fmt.Println(parsedData)

	function, args, err := argsParser.ParseData(string(dataField))
	if err != nil {
		fmt.Println("error", err)
	}

	// Nothing (due to error: tokenize failed)
	fmt.Println(function)
	fmt.Println(args)
}
```

Example of data (BQ):
https://console.cloud.google.com/bigquery?sq=667383445384:515b103ed915424580fb62f75e55c5b5

However, we should maybe use `parsers.NewESDTTransferParser`? Because now, the results parser hardcodes some incomplete logic with `if strings.HasPrefix(data, "ESDTTransfer")` (fragile).

Reply:
I thought so as well. But the `parsers.NewESDTTransferParser` will retrieve more info about the transfer itself than the current `sliceDataFieldInParts`. Was wondering if we can agree upon on a better struct that better reflect ESDTTransfers